### PR TITLE
dev-java/javassist: add 3.29.0

### PR DESCRIPTION
**Pull request by vaukai**
Signed-off-by: Volkmar W. Pogatzki <gentoo@pogatzki.net>

### CODE REVIEW

File: dev-java/javassist/javassist-3.29.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+DEPEND="
+	>=virtual/jdk-11:*
+	test? (
+		dev-java/hamcrest-library:1.3
+	)
+"
+
+# https://github.com/jboss-javassist/javassist/blob/rel_3_29_0_ga/pom.xml#L156-L157
+RDEPEND="
+	>=virtual/jre-11:*
```
Comment (Flowdalic) [inline dev-java/javassist/javassist-3.29.0.ebuild:129]: Isn't the code at the provided link in the comment just saying that the tests require java 11 while the library itself is fine with java 8? In that case, maybe we could make the it conditional on the 'test' use flag, whether or not we require at least java 8 or java 11?

File: dev-java/javassist/javassist-3.29.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+#	6) testLocalVarAttribute(javassist.bytecode.BytecodeTest)
+#	junit.framework.AssertionFailedError
+#		at junit.framework.Assert.fail(Assert.java:55)
+#		at junit.framework.Assert.assertTrue(Assert.java:22)
+#		at junit.framework.Assert.assertTrue(Assert.java:31)
+#		at junit.framework.TestCase.assertTrue(TestCase.java:200)
+#		at javassist.bytecode.BytecodeTest.testLocalVarAttribute(BytecodeTest.java:335)
+#
+#	FAILURES!!!
+#	Tests run: 431,  Failures: 6
```
Comment (Flowdalic) [inline dev-java/javassist/javassist-3.29.0.ebuild:111]: Given that we strive to keep sizes in the ::gentoo tree minimal, as every uses sync this, I find this comment a little bit to verbose, even though I always in favor of code comments. You can definitely place it in the commit messages (as those are not typically synced to the user) and just leave a short comment that you experienced test failures even though with "mvn test" they all pass.

Comment (vaukai) [reply dev-java/javassist/javassist-3.29.0.ebuild:111]: I'll move it to a bug report.  That's easiest to find.

File: dev-java/javassist/javassist-3.29.0.ebuild (hunk 1 above)
Comment (vaukai) [reply dev-java/javassist/javassist-3.29.0.ebuild:129]: Tried that but failed.  Can you show me a working example?

File: dev-java/javassist/javassist-3.29.0.ebuild (hunk 2 above)
Comment (Flowdalic) [reply dev-java/javassist/javassist-3.29.0.ebuild:23]: Even better, just make sure that there is a short comment in the ebuild linking the bug.

Comment (vaukai) [reply dev-java/javassist/javassist-3.29.0.ebuild:23]: > Even better, just make sure that there is a short comment in the ebuild linking the bug.

Done: https://github.com/gentoo/gentoo/pull/26079/files#diff-52f551d46819df9f1e51d739c7f2b0570c32d371ee5ecb469ac768d3667e891bR25

**Pull request closed without merging**